tests/test_class_notice_origin.py

```diff
@@ -0,0 +1,117 @@
+'''
+Copyright (c) 2018 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: BSD-2-Clause
+'''
+
+import unittest
+
+from report import formats
+
+from classes.notice import Notice
+from classes.notice import NoticeException
+from classes.notice_origin import NoticeOrigin
+
+
+class TestClassNoticeOrigin(unittest.TestCase):
+
+    def setUp(self):
+        self.notice_info = Notice("info")
+        self.notice_warn = Notice("warning", "warning")
+        self.notice_errr = Notice("error", "error")
+        self.notice_hint = Notice("hint", "hint")
+
+        self.notices = [
+            self.notice_info,
+            self.notice_warn,
+            self.notice_errr,
+            self.notice_hint
+        ]
+
+        self.notice_origin = NoticeOrigin(origin_str='origin_str')
```

Comment:
I don't think you need to specify the key and value here. `NoticeOrigin('origin_str')` is enough. Only use it when there are preset arguments. Eg: `def options(required, optional=None)` so when using this without the extra argument you will do `options('required')` and with the extra argument`options('required', optional='ok')`

Reply:
I can change that. It's a bit of a force of habit for me. 👍